src/modules/mavlink/streams/ADSB_VEHICLE.hpp

```diff
@@ -103,6 +128,89 @@ class MavlinkStreamADSBVehicle : public MavlinkStream
 			sent = true;
 		}
 
+#if CONFIG_DRIVERS_TRANSPONDER_UAVIONIX
+
+		if (_mavlink->get_mode() == Mavlink::MAVLINK_MODE_UAVIONIX) {
+
+			// Required update for dynamic message is 5 [Hz]
+			mavlink_uavionix_adsb_out_dynamic_t dynamic_msg = {
+				.utcTime = UINT32_MAX,
+				.gpsLat = INT32_MAX,
+				.gpsLon = INT32_MAX,
+				.gpsAlt = INT32_MAX,
+				.baroAltMSL = INT32_MAX,
+				.accuracyHor = UINT32_MAX,
+				.accuracyVert = UINT16_MAX,
+				.accuracyVel = UINT16_MAX,
+				.velVert = INT16_MAX,
+				.velNS = INT16_MAX,
+				.VelEW = INT16_MAX,
+				.state = UAVIONIX_ADSB_OUT_DYNAMIC_STATE_ON_GROUND,
+				.squawk = static_cast<uint16_t>(_adsb_squawk.get()),
+				.gpsFix = 0,
+				.numSats = UINT8_MAX,
+				.emergencyStatus = static_cast<uint8_t>(_adsb_emergc.get())
+			};
+
+			vehicle_status_s vehicle_status;
+
+			if (_vehicle_status_sub.update(&vehicle_status)) {
+				if (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) {
+					dynamic_msg.state |= ~UAVIONIX_ADSB_OUT_DYNAMIC_STATE_ON_GROUND;
+				}
+			}
+
+			sensor_gps_s vehicle_gps_position;
+
+			if (_vehicle_gps_position_sub.update(&vehicle_gps_position)) {
+
+				dynamic_msg.utcTime = static_cast<uint32_t>(vehicle_gps_position.time_utc_usec / 1000000ULL);
+				dynamic_msg.gpsLat = vehicle_gps_position.lat;
+				dynamic_msg.gpsLon = vehicle_gps_position.lon;
+				dynamic_msg.gpsAlt = vehicle_gps_position.alt;
```

Comment:
this should be the WGS84 height, not AMSL: https://github.com/mavlink/mavlink/blob/af35d3a47c3f4d9abbc44f8f5499c047d8802e91/message_definitions/v1.0/uAvionix.xml#L103
https://github.com/PX4/PX4-Autopilot/blob/ce609144b0b8058cf4d05dac691c36c4f0a4ffd0/msg/sensor_gps.msg#L11
```suggestion
				dynamic_msg.gpsAlt = vehicle_gps_position.alt_ellipsoid;
```